SurfaceDevCenterManager/DevCenterApi/DevCenterHandler.cs

```diff
@@ -101,25 +106,46 @@ private string GetDevCenterBaseUrl()
                         }
                     }
 
-                    break;
-                }
+                    string content = await infoResult.Content.ReadAsStringAsync();
+                    if (infoResult.IsSuccessStatusCode)
+                    {
+                        processContent?.Invoke(content);
+                        return null;
+                    }
 
-                string content = await infoResult.Content.ReadAsStringAsync();
-                if (infoResult.IsSuccessStatusCode)
-                {
-                    processContent?.Invoke(content);
-                    return null;
-                }
+                    try
+                    {
+                        reterr = JsonConvert.DeserializeObject<DevCenterErrorReturn>(content);
+                    }
+                    catch (JsonReaderException)
+                    {
+                        // Error is in bad format, return raw
+                        reterr = new DevCenterErrorReturn()
+                        {
+                            HttpErrorCode = (int)infoResult.StatusCode,
+                            StatusCode = infoResult.StatusCode.ToString("D") + " " + infoResult.StatusCode.ToString(),
+                            Message = content
+                        };
+                    }
 
-                DevCenterErrorReturn reterr = JsonConvert.DeserializeObject<DevCenterErrorReturn>(content);
-                // reterr can be null when there is HTTP error
-                if (reterr == null)
-                {
-                    return new DevCenterErrorDetails
+                    // reterr can be null when there is HTTP error
+                    if (reterr == null)
```

Comment:
It's not super clear how/why the json parsing would return a null -- as I would think it would either return a valid reterr or throw an exception. But I'm going to trust the comment on line 131 for now. :-)